### DISCUSSION TITLE
bin/rails s -u puma vs bundle exec puma

### DISCUSSION THREAD

**Discussion by kawadhiya21** [puma/puma#3096] [Q&A] [answered]
Just want to know what are the implications when we use `bundle exec rails s -u puma` instead of `bundle exec puma` that too in production environment. What puma functionalities would be missed out if we do this?

We have a specific use case around log formatting which is forcing us to use `bundle exec rails s -u puma`.

**Comment by nateberkopec** [comment, marked as answer]
Basically not all of our command line options are available from the Rails utility.

I don't think there's a difference if you are loading all of your configuration from a `.rb` file, e.g. `puma.rb`. Anyone want to correct me?

**Reply by stevenharman** [reply]
That's always been my understanding/expectation as well.
